cache-filter/src/filter/http.rs

```diff
@@ -161,8 +161,47 @@ impl Context for CacheFilter {
         match self.get_http_call_response_body(0, body_size) {
             Some(bytes) => {
                 match Authorization::from_str(std::str::from_utf8(&bytes).unwrap()) {
-                    Ok(_response) => {
-                        // Handle cache hit here
+                    Ok(Authorization::Ok(response)) => {
+                        // Form application struct from the response
+                        let key_split = self.cache_key.split('_').collect::<Vec<_>>();
+                        let mut state = HashMap::new();
+                        let UsageReports::UsageReports(reports) = response.usage_reports();
+                        for usage in reports {
+                            state.insert(
+                                usage.metric.clone(),
+                                UsageReport {
+                                    period_window: PeriodWindow {
+                                        // i64 to u64 but value will always be positive
+                                        start: Duration::from_secs(
+                                            usage.period_start.0.try_into().unwrap(),
```

Comment:
Same thing as above, you can use `thiserror`, define your module's error type, change the signature to return a `Result` and do `usage.period_start.0.try_into()?,`.

Reply:
Fixed